### PR TITLE
New chronosuit teleport mechanics

### PR DESCRIPTION
**Pull request by JJRcop**
### Revised edition of chronosuit teleport mechanics:
- You no longer continuously teleport to the camera while moving, you must stop moving for a few moments or press "teleport now" to teleport
- Your teleport is a static amount of time when phasing out, but it takes an amount of time to phase in depending on how far you travel, which is approximately 1 second per 5 tiles
- Your suit still deactivates when strong EMPed like normal, but if you are EMPed while you are in the middle of teleporting, the suit deactivates, but you are also electrocuted for 35 damage
- While using the camera to maneuver to the target location, a ghost icon of you sticks to the center of the screen to help show you which tile you are teleporting to
- New phasing animation

---

### CODE REVIEW

**Comment by PKPenguin321** [comment]
i love the 10,000 emps in that demo video
also, admins should use these more

File: code/modules/clothing/spacesuits/chronosuit.dm [hunk 1]
```diff
@@ -82,52 +109,60 @@
...
+		phase_underlay.animate_movement = NO_STEPS
+		phase_underlay.alpha = 0
+		phase_underlay.mouse_opacity = 0
+		phase_underlay.name = user.name
+		phase_underlay.transform = user.transform
+		phase_underlay.pixel_x = user.pixel_x
+		phase_underlay.pixel_y = user.pixel_y
+
+		user.animate_movement = NO_STEPS
+		user.next_click = world.time + (8 + phase_in_ds)
```
Comment (JJRcop) [inline code/modules/clothing/spacesuits/chronosuit.dm]: i realize this is bad i'm working on it

**Comment by Bawhoppen** [comment]
bigtime nerf to chrono-legionairres, maybe they will see some use now

File: code/modules/clothing/spacesuits/chronosuit.dm [hunk 2]
```diff
@@ -82,52 +109,60 @@
...
+		phase_underlay.master = user
+		phase_underlay.animate_movement = NO_STEPS
+		phase_underlay.alpha = 0
+		phase_underlay.mouse_opacity = 0
+		phase_underlay.name = user.name
+		phase_underlay.transform = user.transform
+		phase_underlay.pixel_x = user.pixel_x
+		phase_underlay.pixel_y = user.pixel_y
+
+		user.animate_movement = NO_STEPS
```
Comment (RemieRichards) [inline code/modules/clothing/spacesuits/chronosuit.dm]: Wow hey wtf are you doing here.
P.sure this doesn't do what you think it does.

**Comment by JJRcop** [comment]
@Bawhoppen Yep, that's the plan!

File: code/modules/clothing/spacesuits/chronosuit.dm (hunk 2 above)
Comment (JJRcop) [inline code/modules/clothing/spacesuits/chronosuit.dm]: making it so you don't slide if you teleport to an adjacent tile, it does do exactly what I think it does

**Comment by JJRcop** [comment]
I fixed the next_click stuff and merge conflicts

**Comment by JJRcop** [comment]
Broken atm, "teleport now" button needs to be modified

**Comment by JJRcop** [comment]
Fixed it

**Comment by JJRcop** [comment]
Aran brought up some points that I should fix, don't merge this yet

**Comment by JJRcop** [comment]
Had some downtime, will work on tomorrow or weekend.

**Comment by JJRcop** [comment]
Mokay I think that's it.

File: code/modules/clothing/spacesuits/chronosuit.dm [hunk 3]
```diff
@@ -25,19 +25,27 @@
 	action_button_name = "Toggle Chronosuit"
 	armor = list(melee = 60, bullet = 60, laser = 60, energy = 60, bomb = 30, bio = 90, rad = 90)
 	var/list/chronosafe_items = list(/obj/item/weapon/chrono_eraser, /obj/item/weapon/gun/energy/chrono_gun)
+	var/list/hands_nodrop_states
```
Comment (RemieRichards) [inline code/modules/clothing/spacesuits/chronosuit.dm]: a list for 2 values is silly.
use a bitfield.

Comment (JJRcop) [inline code/modules/clothing/spacesuits/chronosuit.dm]: ah good idea

**Comment by JJRcop** [comment]
Dee dum, oh, fixed!

**Pull request merged**